### ISSUE TITLE
Integer key in view

### ISSUE THREAD

**Issue by Fulnir** [locolupo/couchdb_connector#40]
When I use a view with an integer key, the function `document_by_key ` doesn't work.

`{:ok, body} = document_by_key(@db_props, %{design: "user", view: "by_s_id", key: Integer.to_string(s_id)}, :ok)`

resulted in an empty list.

The url created by `unauthenticated_document_by_key`  is like this
`http://localhost:5984/db_dev/_design/user/_view/by_s_id?key=\"18155059\"&stale=ok`

In the terminal I played with some queries.
`curl -X GET http://localhost:5984/db_dev/_design/user/_view/by_s_id?key=\"18155059\"&stale=ok`

Don't work. Result rows = empty list

```
curl -X GET http://localhost:5984/db_dev/_design/user/_view/by_s_id?key="18155059"&stale=ok
curl -X GET http://localhost:5984/db_dev/_design/user/_view/by_s_id?key=18155059&stale=ok
```

Both work. rows = One result.

I started this month with Elixir and I don't know if I do something wrong.

I have written a function `document_by_int_key`. This create in query_int_path
`key=18155059` and not `key=\"18155059\"`  for the url

**Comment by mkrogemann** [comment]
Thank you for reporting this. It definitely looks like a bug and I will look into it as soon as possible, likely tomorrow.

**Comment by mkrogemann** [comment]
The handling of keys is somewhat confusing of maybe even buggy? 
With CouchDB 1.6.1, I have to use this url

`http://127.0.0.1:5984/couchdb_connector_test/_design/test_view/_view/test_fetch?key=\"test_name\"&stale=update_after`

To find this document

```
{
   "_id": "test_id",
   "_rev": "1-4e1b5761c76190387d99334f869df218",
   "name": "test_name"
}
```
using this simple view

```
{
   "_id": "_design/test_view",
   "_rev": "1-0aee7b6751b597aea0aa0a9f4f14aa37",
   "views": {
       "test_fetch": {
           "map": "function(doc){emit(doc.name, doc)}"
       }
   }
}
```

But if I have a document with an integer name like this

```
{
   "_id": "test_id",
   "_rev": "1-4e1b5761c76190387d99334f869df218",
   "name": 1234
}
```

Then I must remove the backslashes from the URL like so:

`http://127.0.0.1:5984/couchdb_connector_test/_design/test_view/_view/test_fetch?key="1234"&stale=update_after`

I would say that this is not consistent. It's definitely not what I expect.

I am currently thinking about changing the view_key type definition to not restrict to String. 

`@type view_key :: %{design: String.t, view: String.t, key: String.t}`
instead of 
`@type view_key :: %{design: String.t, view: String.t, key: String.t}`

**Comment by mkrogemann** [comment]
I just released version 0.4.5 with a fix. Please have a look if it fixes your issue. You can now use an integer key in the type view_key without having to convert it to a String first.
